sdk/batch/batch-rest/MigrationGuide.md

```diff
@@ -0,0 +1,265 @@
+# Guide for migrating to `@azure-rest/batch` from `@azure/batch`
+
+This guide is intended to assist customers in the migration to `@azure-rest/batch` from the legacy `@azure/batch` package. It will focus on side-by-side comparisons for similar operations between the two packages.
+
+Familiarity with the legacy client library is assumed. For those new to the Azure Batch JavaScript client library, please refer to [README](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/batch/batch-rest/README.md) and [samples](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/batch/batch-rest/samples) of `@azure-rest/batch` instead of this guide.
+
+## Table of contents
+
+- [Migration benefits](#migration-benefits)
+- [Constructing the clients](#constructing-the-clients)
+  - [Authenticate with shared key credentials](#authenticate-with-shared-key-credentials)
+  - [Authenticate with Microsoft Entra ID](#authenticate-with-microsoft-entra-id)
+- [Operation response differences](#operation-response-differences)
+- [Error handling](#error-handling)
+- [More Examples](#more-examples)
+  - [Create pools](#create-pools)
+  - [Create jobs](#create-jobs)
+  - [Submit tasks](#submit-tasks)
+
+## Migration benefits
+
+- Reduced package sizes, `@azure-rest/batch` comes in a form called Rest Level Client(RLC), which is much more lightwight than the traditional Modular Client like `@azure/batch`. It take advantages of Typescript type inferences and reduce bundle size if you were to use it in a browser environment. For more referece of RLC, please see this [doc](https://github.com/Azure/azure-sdk-for-js/blob/main/documentation/rest-clients.md) and our introduction [blog](https://devblogs.microsoft.com/azure-sdk/azure-rest-libraries-for-javascript/).
```

Comment:
Spelling mistake: "lightwight" -> "lightweight"

Reply:
Suggested edits:

- Reduced package size: `@azure-rest/batch` is a [REST client](https://github.com/Azure/azure-sdk-for-js/blob/main/documentation/rest-clients.md), which is more lightweight compared to the `@azure/batch` package. It take advantages of Typescript type inferences and reduces bundle size when used in a browser environment. For more information, please see this [doc](https://github.com/Azure/azure-sdk-for-js/blob/main/documentation/rest-clients.md) and our introduction [blog](https://devblogs.microsoft.com/azure-sdk/azure-rest-libraries-for-javascript/).